lib/zold/version_file.rb

```diff
@@ -9,11 +9,22 @@ def initialize(path, log: Log::Verbose.new)
       @log = log
     end
 
-    # @todo #285:30min Replace this stub with functionality.
-    #  We need to run the script (`yield`) if the version of
-    #  the script is between the saved version and the current one.
     def apply(version)
-      @log.info("Version #{version} doesn't need to be applied.")
+      return unless Gem::Version.new(version) > current_data_version
+
+      yield version
+    end
+
+    private
+
+    def current_data_version
+      return Gem::Version.new('0') unless File.exist?(version_file)
+
+      @current_data_version ||= Gem::Version.new(File.read(version_file))
```

Comment:
@ledestin this will have "\n" in it. Are you sure `Gem::Version` Is OK with it? I'd be rather surprised.

Reply:
@botanicus thanks, updated